components/gitpod-cli/pkg/gitpod/server.go

```diff
@@ -50,10 +58,15 @@ func ConnectToServer(ctx context.Context, wsInfo *supervisor.WorkspaceInfoRespon
 	if err != nil {
 		return nil, xerrors.Errorf("failed getting token from supervisor: %w", err)
 	}
+
 	client, err := serverapi.ConnectToServer(wsInfo.GitpodApi.Endpoint, serverapi.ConnectToServerOpts{
 		Token:   clientToken.Token,
 		Context: ctx,
 		Log:     log.NewEntry(log.StandardLogger()),
+		ExtraHeaders: map[string]string{
```

Comment:
I made it so that all connections with the CLI server-connecting function get these headers, because I think it is good info to have from all places that call it.

Reply:
great idea 

Did you manage to make version resolution work? As far as I understand it should be inlined in the file during the build. Please take time to understand how it happens in supervisor build.